packages/vite/src/index.ts

```diff
@@ -225,3 +229,30 @@ const applyOn = (apply: 'build' | 'serve') => {
     }
   }
 }
+
+// @MARK
+// Currently only used in testing
+export const vitePrebuildWebFile = async (
+  srcPath: string,
+  flags: Flags = {}
+) => {
+  const code = await transform(srcPath)
+  const config = getWebSideDefaultBabelConfig(flags)
+  const result = babel.transform(code, {
+    ...config,
+    cwd: getPaths().web.base,
+    filename: srcPath,
+  })
+
+  return result
+}
+
+export const transform = async (srcPath: string) => {
+  const code = readFileSync(srcPath, 'utf-8')
+
+  const transformed = await transformWithEsbuild(code, srcPath, {
+    loader: 'jsx',
+  })
+
+  return transformed.code
+}
```

Comment:
@Tobbe per our discussion yesterday, let's move this and its dependency into the tests file so that we don't publish it and add extra weight to the module